packages/ui/components/head-seo/HeadSeo.tsx

```diff
@@ -1,13 +1,16 @@
 import type { NextSeoProps } from "next-seo";
 import { NextSeo } from "next-seo";
 import { usePathname } from "next/navigation";
+import { useEffect, useState } from "react";
 
 import type { AppImageProps, MeetingImageProps } from "@calcom/lib/OgImages";
 import { constructAppImage, constructGenericImage, constructMeetingImage } from "@calcom/lib/OgImages";
 import { APP_NAME, CAL_URL } from "@calcom/lib/constants";
 import { buildCanonical, getSeoImage, seoConfig } from "@calcom/lib/next-seo.config";
 import { truncateOnWord } from "@calcom/lib/text";
 
+import text from "../../../../apps/web/public/static/locales/en/common.json";
```

Comment:
wouldn't this cause the locales to be downloaded twice? one in the bundle and one in the next.js props?

Reply:
I added a bit more context on the PR description but tldr yes it would however I think the size is negligible compared to the problems it solves. With RSC the story is a bit different because you can technically stream compute from the server so would allow you to stream translation from the server directly however I think that would be a far bigger effort. What I also suggested in the description is that if there is interest to adopt this pattern, what we can do is on the `useLocale` hook that is used everywhere add the english translations as default and when the locale of a user is english avoid fetching the translation file again so this would result on a net zero impact for english users and yes it would have a cost for other users but again, I think the benefits outweighs the costs. What you think @zomars? 

Eg this is what people see when they search for "cal com sign up"
<img width="641" alt="Screenshot 2024-06-07 at 4 29 43 PM" src="https://github.com/calcom/cal.com/assets/70239736/cbdcd1b2-0278-4fcd-9a6a-5b07fbee83a7">